rebasebot/bot.py

```diff
@@ -163,6 +174,16 @@ def _do_rebase(gitwd, source, dest, source_repo, tag_policy):
             if not _resolve_rebase_conflicts(gitwd):
                 raise RepoException(f"Git rebase failed: {ex}") from ex
 
+    if update_go_modules:
+        # We have a list of all the bot commits with go mod updates
+        # We reset all these commits and delete them
+        # The flag `update_go_modules` will create new go mod updates commit
+        for commit in go_mod_commits:
+            gitwd.git.cherry_pick(f"{sha}", "-Xtheirs")
+        gitwd.git.reset("--hard", f"HEAD~{len(go_mod_commits)}")
+        gitwd.git.commit("-m", "UPSTREAM: <carry>: Updating and vendoring go modules "
+                "after an upstream rebase")
```

Comment:
So this squashes all of the existing go mod updates right? Why not just drop them completely if a new one is being created later? If we keep this, we will end up with 2 commits for go modules right?

Reply:
No, this `deletes` all the previous commits - until we create a new one, there are none of these (with the flag `update_go_modules`, we are creating it anyways).

Reply:
yeah, i'm not following this, it seems like we cherry pick the commit and the reset the head back to before the picks. shouldn't we just not do the cherry picks or the reset?

or are we doing the cherry-picks to ensure they already exist in the commit history?

Reply:
But you are creating a commit at the end of this, surely if we didn't have this loop, we would just drop all the go mod update commits (as they are seen as `<drop>`) and then the other part of the `update_go_modules` that runs the update handles creating the new commit?